cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -41,40 +43,52 @@ def domain(request):
     return dom
 
 
-@pytest.fixture(scope="module")
-def test_vm(setup_provider_modscope, provider, vm_name, request):
-    """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = VM.factory(vm_name, provider, template_name=provider.data['full_template']['name'])
+def provision_vm(provider, vm_name, request, small_template=True):
+    if small_template:
+        vm = VM.factory(vm_name, provider, template_name=provider.data['small_template'])
```

Comment:
This requires change in the structure of yaml file if I understand your request correctly. Probably affecting a lot of stuff. It is not my ambition to change lot of stuff in this PR, I just need to change bits of snapshot testing.